From 1e74b7eebe68b8428aab0e397efa438b3efcb970 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Mon, 15 Jun 2020 14:30:09 +0800
Subject: [PATCH] ARM: dts: rockchip: fix trust shared memory mapping for
 rv1126-evb-ddr3-v10-tb

Allow kernel access 0x00200000~0x00208000, because the first usable region
must be PMD aligned. Otherwise rampoops would not work.

Fixes: 9e109a260083 ("ARM: dts: rockchip: reserve 32KB shared memory for rv1126-evb-ddr3-v10-tb")
Change-Id: I170a6d946695dbace8064eafa57986787d8cab86
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 arch/arm/boot/dts/rv1126-evb-ddr3-v10-tb.dts | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/rv1126-evb-ddr3-v10-tb.dts b/arch/arm/boot/dts/rv1126-evb-ddr3-v10-tb.dts
index dd4401da96ab..acdf86ea8ecc 100644
--- a/arch/arm/boot/dts/rv1126-evb-ddr3-v10-tb.dts
+++ b/arch/arm/boot/dts/rv1126-evb-ddr3-v10-tb.dts
@@ -25,10 +25,14 @@
 
 	reserved-memory {
 		trust@0 {
-			reg = <0x00000000 0x00208000>;
+			reg = <0x00000000 0x00200000>;
 			no-map;
 		};
 
+		trust@200000 {
+			reg = <0x00200000 0x00008000>;
+		};
+
 		mmc_ecsd: mmc@20f000 {
 			reg = <0x0020f000 0x1000>;
 		};
-- 
2.35.3

